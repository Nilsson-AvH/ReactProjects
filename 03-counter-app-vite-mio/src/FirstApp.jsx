import PropTypes  from 'prop-types'
// const newMessage = {
  //     message: 'Hola Mundo',
  //     title: 'Nilson'
  // }
  
  const getResult = () => {
    return ({
      // <div>FirstApp</div>
      message: 'Hola Mundo',
      title: 'El gato',
      text: 'internal test text',
    })
  }
  

const FirstApp = ( { title, subTitle, name } ) => {
  
  //console.log ( props )
  
    /* if ( !title ) {
      throw new Error('El title no existe.');
    } */
    // Para evitar colocar todo este codigo (el if) se usan las PropTypes en este caso para validar que no exista un titulo.

    return (        
        <> {/* Esta es la abreviaci√≥n de Fragment: <></> */}

        <h1> { title } </h1>

        <code> { JSON.stringify( getResult() ) } </code>

        {/* el JSON.stringify() me ayuda a imprimir una funcion en este caso con multiples 
        "cosas"  dentro y el <code></code> es simplemtente para que se vea mas bonito,
        facilmente puede ir un <h1></h1> o un <p></p>*/}

        <p> { subTitle } </p>
        <p> { name } </p>
        
        </>

    )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired, //este PropTypes me indica que title deber ser string y es requerido y/o/u obligatorio
  subTitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  // title: 'No hay titulo',
  subTitle: 'No hay subtitulo',
  name: 'Nilson LoRito',
}

export default FirstApp