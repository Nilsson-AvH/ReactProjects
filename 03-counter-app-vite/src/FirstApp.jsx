// const newMessage = {
//     message: 'Hola Mundo',
//     title: 'Nilson'
// }

const getResult = () => {
  return ({
    // <div>FirstApp</div>
    message: 'Hola Mundo',
    title: 'El gato',
    text: 'internal test text',
  })
}


const FirstApp = ( { title, subTitle } ) => {
    
    //console.log ( props )

    // No es recomentable que este valor este dentro de este functional component
    // es mejor sacarlo aunque eso no significa que vaya a ser un valor dentro del scope global,
    // por que el queda encapsulado dentro de FirstApp.jsx

    return (

        
        <> {/* Esta es la abreviaci√≥n de Fragment: <></> */}

            <h1> { title } </h1>

            {/* <code>First App by { JSON.stringify( newMessage ) } </code> */}

            <code> { JSON.stringify( getResult() ) } </code> 

            {/* el JSON.stringify() me ayuda a imprimir una funcion en este caso con multiples 
            "cosas"  dentro y el <code></code> es simplemtente para que se vea mas bonito,
            facilmente puede ir un <h1></h1> o un <p></p>*/}

            <p> { subTitle + 7 } </p>
        </>
    )
}

export default FirstApp