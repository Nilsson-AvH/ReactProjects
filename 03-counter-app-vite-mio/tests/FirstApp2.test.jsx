import { getAllByAltText, getAllByText, render, screen } from "@testing-library/react"
import FirstApp from "../src/FirstApp"

describe('Prueba del archivo FirstApp.jsx', () => { 
   
  const title = 'Hola soy Goku';
  const subTitle = 'Hola soy un subtitulo.';

    test('Debe hacer match con el snapshot.', () => { 
        const {container} = render( <FirstApp title={title} /> );
        // console.log(container);
        expect(container).toMatchSnapshot();
     })

     test('Debe mostrar el mensaje "Hola soy Goku"', () => {
      
       render( <FirstApp title={title}/> );
       expect(screen.getAllByText(title)).toBeTruthy();
       
      //  screen.debug();
      })
    test('Debe mostrar el titulo en un h1', () => { 
      render( <FirstApp title={title}/> );
      console.log( screen.getByRole('heading',{level:1}).innerHTML );
      expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
     })

     test('Debe mostrar el subtitulo enviado por props.', () => { 
      render( < FirstApp title={title} subTitle={subTitle} /> );
      expect( screen.getAllByText(subTitle).length ).toBe(1);

      })
    
 })