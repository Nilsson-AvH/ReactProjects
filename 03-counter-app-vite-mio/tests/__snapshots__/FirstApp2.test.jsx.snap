// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del archivo FirstApp.jsx Debe hacer match con el snapshot. 1`] = `
<div>
   
  <h1
    data-testid="test-title"
  >
     
    Hola soy Goku
     
  </h1>
  <code>
     
    {"message":"Hola Mundo","title":"El gato","text":"internal test text"}
     
  </code>
  <p>
     
    No hay subtitulo
     
  </p>
  <p>
     
    Nilson LoRito
     
  </p>
</div>
`;
